resources/app/src/components/expert_panels/ReviewMembership.vue

```diff
@@ -20,152 +20,190 @@
         fields.value.push('coi_completed');
     }
 
-    const members = ref([]);
-    const chairs = computed(() => tableRows.value.filter(m => m.roles.includes('chair')));
-    const experts = computed(() => tableRows.value.filter(m => m.roles.includes('expert')));
-
-    const memberGroups = computed(() => [
-        {
-            title: 'Leadership',
-            members: chairs.value
-        },
-        {
-            title: 'Coordination',
-            members: tableRows.value.filter(m => m.roles.includes('coordinator'))
-        },
-        {
-            title: 'Biocuration',
-            members: tableRows.value.filter(m => m.roles.includes('biocurator'))
-        },
-        {
-            title: 'Expertise',
-            members: experts.value
-        }
-    ])
-
-    const delay = (ms = 1000) => new Promise((r) => setTimeout(r, ms));
-    const getPublications = async member => {
-        const baseUri = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils';
-        const searchUrl = `${baseUri}/esearch.fcgi?db=pubmed&term=${member.last_name},+${member.first_name}[author]&retmode=json&retmax=0`;
-
-        return axios.get(searchUrl)
-            .then(rsp => {
-                member.pubCount = rsp.data.esearchresult.count
-            })
-            .catch(async error => {
-                if (error.response.status == 429) {
-                    await delay();
-                    getPublications(member);
-                }
-            });
+   // const members = ref([]);
+
+
+   const members = ref([]);
+
+// Watch for changes in props.members and apply sorting and aggregation
+watch(() => props.members, async (to) => {
+    if (!to) {
+        return;
     }
 
-    watch(() => props.members, async to => {
-        if (!to) {
-            return;
+    // Copy the incoming members array
+    members.value = [...to];
+
+    // Sort members based on roles
+    members.value.sort((a, b) => {
```

Comment:
At first glance, it might make more sense to decorate each member object with a sort key based on the priority specified.

So when loading the members array, something like:
```
function sort_priority(roles) {
  const priorities = ['chair', 'coordinator', 'expert', 'coordinator']
  return Math.min(roles.map( x => priorities.indexOf(x) ))
}

members.forEach(m => m.sort_priority = sort_priority(m.roles)
```

Then you could just sort with members.sort_priority as the sort key.

Reply:
Hmm... now I see you have something like this below, so is this redundant code that should be removed?